### ISSUE TITLE
[Bug]: Discord Activity Status not clearing upon pause

### ISSUE THREAD

**Issue by SoNothingMC** [ciderapp/Cider-2#224]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://cider.sh/legal/conduct) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ciderapp/Sabiiro-BugTracker/issues) for a bug report that matches the one I want to file, without success.
- [X] I have read the [troublshooting FAQs](https://cider.gitbook.io/welcome-to-gitbook/support/faqs) and none solved my issue.

### What operating system are you using?

Windows

### Build Details

Sabiiro 2.0.3 —  Built on 27/5/2023, 1:02:04 am (UTC+8)

### Operating System Version

Windows 10 22H2 (19045.2965)

### Describe the Bug

*This is a different issue as #28.*

When pausing or stopping a song, the Discord Activity Status still shows Cider 2 as an active game, even if the user has "Clear Discord Rich Presence on Pause" enabled.

![The Discord UI, with Playing Apple Music text shown.](https://github.com/ciderapp/Sabiiro-BugTracker/assets/32032824/bec59981-ab82-4063-b29d-ab7c781f8fb6)
*Now Playing still active when track is paused/stopped.*

### Steps to Reproduce

1. Enable "Clear Discord Rich Presence on Pause" in Settings → Connectivity 
2. Play a track
3. Pause the track
4. See result

### Anything else?

No, this is not related to *anything* in the Discord User Settings menu.

and no, this is also not Cider ignoring the settings option, since if I have the option disabled (i.e. Show Rich Presence even if paused), it will be showing a Rich Presence and displaying the paused track's details and album art.

### Build timestamp

1685542397555

**Comment by coredev-uk** [comment]
This is an issue with Sabiiro's implementation of DiscordRPC. I believe this is supposed to be the idle status, which Electron does not do. I will forward it onto the team to get this changed.

**Comment by CharliDaiz** [comment]
Are they still looking into it?

**Comment by Natron4** [comment]
> Are they still looking into it?

I hope so lol, it's been a while

**Comment by d3rpp** [comment]
Just and update on the behaviour 

On windows this option entirely disconnects the RPC client, at which point discord can take up to 20 seconds to update.

This is done with the intention of allowing other things to take over, but we're actively looking into making this faster if we can to reduce this exact issue's effect

**Comment by Natron4** [comment]
Wait sorry lol what does "Just and update on the **behaviour" mean?

> Just and update on the behaviour
> 
> On windows this option entirely disconnects the RPC client, at which point discord can take up to 20 seconds to update.
> 
> This is done with the intention of allowing other things to take over, but we're actively looking into making this faster if we can to reduce this exact issue's effect

**Comment by cryptofyre** [comment]
Mostly intended behavior due to the timeout between Sabiiro (backend) actually disconnecting from the IPC thread.

**Comment by Natron4** [comment]
Oh hell no.  Closed??  This is a HUGE flaw guys.  If I'm not listening to anything I want my status to be gone.  There's an option in the settings for a reason, yet it does not work.

Here's a solution:  Allow us to clear the currently selected song, then nothing will get sent to Discord.  This should happen automatically after finishing an album but even if that specifically doesn't get implemented... create a button to clear the queue INCLUDING the current song.  This will mimic a fresh-opened app, ending the status on Discord.

TLDR: Create a clear queue option

**Comment by d3rpp** [comment]
> Oh hell no. Closed?? This is a HUGE flaw guys. If I'm not listening to anything I want my status to be gone. There's an option in the settings for a reason, yet it does not work.
> 
> Here's a solution: Allow us to clear the currently selected song, then nothing will get sent to Discord. This should happen automatically after finishing an album but even if that specifically doesn't get implemented... create a button to clear the queue INCLUDING the current song. This will mimic a fresh-opened app, ending the status on Discord.
> 
> TLDR: Create a clear queue option

works in theory, but in practice, Discord has this thing called "margin of error" where if you were to disconnect the RPC it assumes an error has occurred. unless the calling process has exited (hence how games do it instantly).

so it waits and just keeps it as it is for a grace period before realising it just disconnected, this is not a flaw in Cider or a flaw in Discord, but it is out of the scope of the Discord RPC.

the current solution clears the RPC, but since it's connected it still says "Listening to ..." or "Playing ...", solution: disconnect, but when we go to do that it assumes failure and doesn't update for a bit for the aforementioned reason.

if a solution that you suggested was simple we would've implemented it, but it's not, this is by design and we can't exactly change how Discord is designed.

Looking at your profile I see you wrote an AM Discord RPC program in Python, taking a look you appear to have a similar issue in that if it clears you get the phantom margin or error or even no result at all. 

This isn't fixed because it is out of the scope of the Discord RPC which was originally designed for games, assuming that disconnecting after closing is OK and disconnecting while still running is an error is a safe assumption by Discord.

Finally, you don't know our codebase, that solution wouldn't work lol.

Sincerely,
the dev that made Sabiiro Discord RPC :heart:
